src/store/notificationsSlice.ts

```diff
@@ -72,3 +74,34 @@ export const showNotification = (payload: Omit<Notification, 'id' | 'timestamp'>
 export const selectNotifications = (state: RootState): NotificationState => {
   return state[notificationsSlice.name]
 }
+
+export const DISMISS_NOTIFICATION_KEY = 'dismissUpdateSafe'
+
+export const notificationsMiddleware: Middleware<{}, RootState> = (store) => (next) => (action) => {
+  const result = next(action)
+
+  switch (action.type) {
+    case notificationsSlice.actions.closeNotification.type: {
+      const state = store.getState()
+      const safe = selectSafeInfo(state)
+
+      const { payload } = action as ReturnType<typeof notificationsSlice.actions.closeNotification>
+      const isUpdateSafeNotification = payload.groupKey === 'safe-outdated-version'
```

Comment:
I'd suggest you export this from the notification file.

Reply:
I moved it into the hook since the constant is being used there as well.